ci/README.md

```diff
@@ -0,0 +1,353 @@
+artifacts, as shown below.
+
+Unfortunately GitHub doesn't give us a link that we can download those from
+inside the VM, but we can download them on our local box, and then scp the
+tarball to the newly created VM that will be used for debugging purposes.
+
+> [!NOTE]
+> Those artifacts are only available when all jobs are finished.
+
+Once you have the kata-static.tar.xz in your VM, you can login to the VM with
+kcli ssh debug-nerdctl-pr8070,go ahead and then clone your development branch
+
+```bash
+$ git clone --branch feat_add-fc-runtime-rs https://github.com/nubificus/kata-containers
+Cloning into 'kata-containers'...
+remote: Enumerating objects: 93672, done.
+remote: Counting objects: 100% (9/9), done.
+remote: Compressing objects: 100% (8/8), done.
+remote: Total 93672 (delta 3), reused 5 (delta 1), pack-reused 93663
+Receiving objects: 100% (93672/93672), 93.72 MiB | 9.52 MiB/s, done.
+Resolving deltas: 100% (60179/60179), done.
+```
+
+Add the upstream as a remote, set up your git, and rebase your branch atop of the upstream's main one
+
+```bash
+$ git remote add upstream https://github.com/kata-containers/kata-containers
+$ git remote update
+$ git config --global user.email "you@example.com"
+$ git config --global user.name "Your Name"
+$ git rebase upstream/main 
+```
+
+Now copy the kata-static.tar.xz into your kata-containers/kata-artifacts directory
+
+```bash
+$ mkdir kata-artifacts
+$ cp ../kata-static.tar.xz kata-artifacts/
+```
+
+> [!NOTE]
+> If you downloaded the .zip from Github you need to uncompress first to see kata-static.tar.xz
+
+And finally run the tests following what's in the yaml file for the test you're
+debugging. 
+
+In our case, the run-nerdctl-tests-on-garm.yaml.
+
+When looking at the file you'll notice that some environment variables are set,
+such as `KATA_HYPERVISOR`, and should be aware that, for this particular example,
+the important steps to follow are:
+
+Install the dependencies
+Install kata
+Run the tests
+
+Let's now run the steps mentioned above exporting the expected environment variables
+
+```bash
+$ export KATA_HYPERVISOR=dragonball
+$ bash ./tests/integration/nerdctl/gha-run.sh install-dependencies
+$ bash ./tests/integration/nerdctl/gha-run.sh install-kata
+$ bash tests/integration/nerdctl/gha-run.sh run
+```
+
+And with this you should've been able to reproduce exactly the same issue found
+in the CI, and from now on you can build your own code, use your own binaries,
+and have fun debugging and hacking! 
+
+### Debugging a kubernetes test
+
+Steps for debugging the kubernetes tests are very similar to the ones for
+debugging non-kubernetes tests, with the caveat that what you'll need, this
+time, is not the kata-static.tar.xz tarball, but rather a payload to be used
+with kata-deploy.
+
+In order to generate your own kata-deploy image you can generate your own
+kata-static.tar.xz and then take advantage of the following script.  Be aware
```

Comment:
Script is https://github.com/kata-containers/kata-containers/blob/main/tools/packaging/kata-deploy/local-build/kata-deploy-build-and-upload-payload.sh .

Reply:
fixed.